src/main/java/com/fasterxml/jackson/databind/jsontype/impl/TypeNameIdResolver.java

```diff
@@ -80,7 +82,16 @@ public static TypeNameIdResolver construct(MapperConfig<?> config, JavaType base
     @Override
     public String idFromValue(Object value)
     {
-        Class<?> cls = _typeFactory.constructType(value.getClass()).getRawClass();
+       return idFromClass(value.getClass());
+    }
+
+
+    protected String idFromClass(Class<?> clazz)
+    {
+        if(clazz==null){
```

Comment:
Would this actually work? Caller assumes `value` is not null by calling `value.getClass()`? Is this method called directly with null from some place?
Alternatively call above could pass null for null instance.